models/src/main/scala/coop/rchain/models/rholang/implicits.scala

```diff
@@ -316,7 +320,9 @@ object implicits {
         case EAndBody(EAnd(p1, p2))     => p1.get.connectiveUsed || p2.get.connectiveUsed
         case EOrBody(EOr(p1, p2))       => p1.get.connectiveUsed || p2.get.connectiveUsed
         case EMethodBody(e)             => e.connectiveUsed
-        case ExprInstance.Empty         => false
+        case EMatchesBody(EMatches(process, pattern)) =>
+          process.connectiveUsed || pattern.connectiveUsed
```

Comment:
No, this is still wrong.

Reply:
So the `connectiveUsed` and `locallyFree` should be only calculated from the `process` (target) and not the `pattern` as well?

Reply:
what would be a rholang test that will fail because of this?